### DISCUSSION TITLE
Public Tag Views

### DISCUSSION THREAD

**Discussion by rasulkireev** [ArchiveBox/ArchiveBox#1047] [Ideas]
I tried finding this in Discussions and in Issues but nothing similar, please let me know if I missed something.

I think it would be insanely useful if users could share urls with posts only under specific Tags. Write now the public page is located at `{url}/public/`. I'm thinking that it would be good to add something like: `{url}/public/tag/{name of the tag}`.

Looking at the code it should be to complicated to do. I can try creating a PR to do that. The questions is, will community find this useful?

**Comment by pirate** [comment]
The URL already supports query parameters for filtering e.g. `/public/?q=search+term+here` which will search tag names as well.

The relevant filtering code is here: https://github.com/ArchiveBox/ArchiveBox/blob/dev/archivebox/core/views.py#L225

Is that enough for your use case?

If not we could consider adding support for more query parameters other than `?q=search+term`, e.g. `?tags__id__exact=30`, `created__gt=2010-10-10`, etc. but I'd rather not unless we really need it.
If we do add support for more I'd rather add support for all the model keys automatically using some built in django features, that would give us support for every query that the admin snapshot list page has (like `?tags__id__exact=...`, `?title__icontains=...`, etc.).

**Reply by rasulkireev** [reply]
I think this should be enough for my use case. Thanks for the prompt reply.

**Comment by wikispecialists** [comment]
Thanks for sharing this with us. I like this. It's much more informative. This ought to be sufficient for my use case. Thank you for your quick response. [wiki specialists](https://wikiparagon.com/)
